presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -0,0 +1,145 @@
+    {
+        try {
+            installedPackageType = PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase());
+        }
+        catch (IllegalArgumentException e) {
+            return Response.status(BAD_REQUEST).entity(text("Bad package format")).build();
+        }
+        try {
+            URL url = new URL(locationToFetchPackage);
+            new Thread(new PrestoCommandAsynchronous(new PrestoInstaller(installedPackageType, url, disableDependencyChecking))).start();
+        }
+        catch (MalformedURLException e) {
+            return Response.status(BAD_REQUEST).entity(text("Invalid url")).build();
+        }
+        return Response.status(ACCEPTED).entity(text("Check status to make sure that presto is installed")).build();
+    }
+
+    @POST
+    @Consumes({"text/plain"})
+    @Produces({"text/plain"})
+    @ApiOperation(value = "Upgrade presto")
+    @ApiResponses(value = {
+            @ApiResponse(code = 202, message = "Acknowledged request"),
+            @ApiResponse(code = 400, message = "Bad package format or Invalid url")
+    })
+    public synchronized Response upgrade(@QueryParam("disableDependencyChecking") @DefaultValue("false") boolean disableDependencyChecking,
+            String locationToFetchPackage)
+    {
+        try {
+            if (installedPackageType != PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase())) {
+                return Response.status(BAD_REQUEST).entity(text("Provided package format doesn't match the installed package format")).build();
+            }
+        }
+        catch (IllegalArgumentException e) {
+            return Response.status(BAD_REQUEST).entity(text("Bad package format")).build();
+        }
+        try {
+            final URL url = new URL(locationToFetchPackage);
+            new Thread(new PrestoCommandAsynchronous(new PrestoUpgrader(installedPackageType, url, disableDependencyChecking))).start();
+        }
+        catch (MalformedURLException e) {
+            return Response.status(BAD_REQUEST).entity(text("Invalid url")).build();
+        }
+        return Response.status(ACCEPTED).entity(text("Check status to make sure that presto is upgraded")).build();
+    }
+
+    @DELETE
+    @ApiOperation(value = "Uninstall presto")
+    @ApiResponses(value = {
+            @ApiResponse(code = 202, message = "Acknowledged request")
+    })
+    public synchronized Response uninstall(@QueryParam("disableDependencyChecking") @DefaultValue("false") boolean disableDependencyChecking,
+            @QueryParam("ignoreNotInstalled") @DefaultValue("false") boolean ignoreNotInstalled)
+    {
+        new Thread(new PrestoCommandAsynchronous(new PrestoUninstaller(installedPackageType, disableDependencyChecking, ignoreNotInstalled))).start();
+        return Response.status(ACCEPTED).build();
+    }
+
+    public enum PackageType
+    {
+        RPM, TARBALL;
+    }
+
+    public class PrestoCommandAsynchronous
```

Comment:
Call this PrestoCommandRunnable (or PrestoAsynchronousCommand... the main thing is that you end up with a noun).